readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -151,14 +155,15 @@ def get_config_params(self):
 
     def append_conf(self, **__):
         """Find or create a ``conf.py`` with a rendered ``doc_builder/conf.py.tmpl`` appended"""
-        try:
-            self.version.get_conf_py_path()
-        except ProjectConfigurationError:
+        if self.config.sphinx.configuration is None:
             master_doc = self.create_index(extension='rst')
             self._write_config(master_doc=master_doc)
 
         try:
-            outfile_path = self.project.conf_file(self.version.slug)
+            outfile_path = (
+                self.config.sphinx.configuration or
+                self.project.conf_file(self.version.slug)
+            )
```

Comment:
It seems here we're implying that if you set a sphinx configuration file, but it doesn't exist, we just create it? I feel like we should error out if you set up a config file and it doesn't actually exist, it will be confusing to the user for the build to pass in this case.

Reply:
This is to keep compatibility with the behavior when the user doesn't set a config file and rtd tries to find one.

Reply:
That part I understand, but this seemed to be creating a 3rd option, where the config was specified but not found, and therefore the config was generated instead. This might not be an issue, I interpreted `self.config.sphinx.configuration` as defaulting to `None` if the file is specified but doesn't exist, but that doesn't seem to be the logic used looking at it again.